script.js

```diff
@@ -311,161 +322,200 @@ function paintScreen() {
-        playerX += 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.x += 10;
-        }
+        changeInX +=10;
     }
+    playerY += changeInY;
+    playerX += changeInX;
     for (let i in items) {
-        if (items[i] == playerIsCarrying) {
+        if (items[i] == playerIsCarrying && playerIsCarrying != null) {
+            playerIsCarrying.x += changeInX;
+            playerIsCarrying.y += changeInY;
             items[i].x = playerIsCarrying.x;
             items[i].y = playerIsCarrying.y;
             items[i].element.style.left = items[i].x + 'px';
-            items[i].element.style.top = items[i].y; + 'px';
+            items[i].element.style.top = items[i].y + 'px';
         }
     }
+    for (let c in containers){
+        if (containers[c] == playerIsCarrying && playerIsCarrying != null) {
+            playerIsCarrying.x += changeInX;
+            playerIsCarrying.y += changeInY;
+            containers[c].x = playerIsCarrying.x;
+            containers[c].y = playerIsCarrying.y;
+            containers[c].element.style.left = containers[c].x + 'px';
+            containers[c].element.style.top = containers[c].y + 'px';
+            for (let i of containers[c].itemIndexInBin){
+                var itemInThisBin = items[i];
+                itemInThisBin.x += changeInX;
+                itemInThisBin.y += changeInY;
+                itemInThisBin.element.style.left = itemInThisBin.x + 'px';
+                itemInThisBin.element.style.top = itemInThisBin.y + 'px';
+
+            }
+            
+        }  
+    }
     player.style.top = playerY + 'px';
     player.style.left = playerX + 'px';
+    
 }
 document.addEventListener('keypress', pickUpItem);
+document.addEventListener('keypress', (event) => {
+    if (event.code == "KeyP") { // Example: Enter key
+        pickUpBin(event);
+    }
+});
+function pickUpBin(event){ 
+    loop3:
+    for (let i in containers) {
+        var thisBin = containers[i];
+        var playerReferencePointX = playerX + player.offsetWidth/2; //center of the head
+        var playerReferencePointY = playerY + player.offsetWidth / 2;
+
+        var distance = [
+            Math.sqrt((Math.pow(Math.abs(playerReferencePointX - thisBin.x), 2)) + (Math.pow(Math.abs(playerReferencePointY - thisBin.y), 2))),
+            Math.sqrt((Math.pow(Math.abs(playerReferencePointX - (thisBin.x + thisBin.element.offsetWidth)), 2)) + (Math.pow(Math.abs(playerReferencePointY - thisBin.y), 2))),
+            Math.sqrt((Math.pow(Math.abs(playerReferencePointX - (thisBin.x + thisBin.element.offsetWidth)), 2)) + (Math.pow(Math.abs(playerReferencePointY - (thisBin.y + thisBin.element.offsetHeight)), 2))),
+            Math.sqrt((Math.pow(Math.abs(playerReferencePointX - thisBin.x), 2)) + (Math.pow(Math.abs(playerY - (thisBin.y + thisBin.element.offsetHeight)), 2))),
+        ];
+        loop4:
+        for ( let dist of distance){
```

Comment:
I've never seen this syntax with the loop naming & break...

I assume this works but it looks like you might just be able to `return` from the function instead of `break`ing out of `loop3`. That would probably be a bit simpler, I think.